src/coreclr/jit/utils.cpp

```diff
@@ -3436,6 +3436,11 @@ uint32_t BitOperations::Log2(uint64_t value)
 // Return Value:
 //    The population count (number of bits set) of value
 //
+#if defined(_MSC_VER)
+// Disable optimizations for PopCount to avoid the compiler from generating intrinsics
+// not supported on all platforms. 
+#pragma optimize("", off)
```

Comment:
Have we root caused and found a minimal repro for this?

Given popcnt is not a x64 baseline instruction, it should not be getting generated without explicit opt in (meaning either we have a cmake/config issue on our end or MSVC has a bug)

Reply:
If it's an MSVC issue, we should ensure an appropriate issue is filed against the compiler to track them fixing this.

Alternatively, if Windows has changed requirements at all, we should ensure it's documented and appropriately take advantage of these requirements (I vaguely remember seeing some change where Windows 11 may require SSE4.2 instructions)

Reply:
> Alternatively, if Windows has changed requirements at all, we should ensure it's documented and appropriately take advantage of these requirements (I vaguely remember seeing some change where Windows 11 may require SSE4.2 instructions)

IIRC next update of Windows 11 requires `popcnt`, however I'd say that dotnet shouldn't require it as it still supports 10 and Server 2012.

Reply:
we should be able to build a repro. I have verified within the debugger that the compiler does generate popcnt, and as Egor points out above looks like a fix has been made recently.

Reply:
If it has been fixed already, you should be able to put the workaround under `_MSC_VER` or `_MSC_FULL_VER` ifdef to make self-documenting and self-healing.

Reply:
yeah checking if we can just pickup the tool update. 

> IIRC next update of Windows 11 requires popcnt, however I'd say that dotnet shouldn't require it as it still supports 10 and Server 2012.

Correct Win11 24H2 is expected to require it, but yeah that is a separate discussion.

Reply:
> If it has been fixed already, you should be able to put the workaround under `_MSC_VER` or `_MSC_FULL_VER` ifdef to make self-documenting and self-healing.

makes sense, but we will need to check the exact versions which had the issue (since it affected both 17.8 and 17.9).

Reply:
According to the links I shared:

>  has been fixed in 17.8.6 and 17.9.0 Preview 5

Reply:
since this is a temporary change, not worth special casing versions in case we get the numbers wrong. The toolset update is going to be in the next few weeks and I can create a revert PR.